### DISCUSSION TITLE
question about the segment threshold

### DISCUSSION THREAD

**Discussion by litun-fkby** [etal/cnvkit#996] [General]
Hi,  i used the cnvkit.py segment to generate the cns file form cnr，and to call the cnv number .  So i have some question about the segment.
The cnvkit version is 0.9.9
The segment command is :
cnvkit segment sample.fix.cnr -p 20s --drop-low-coverage -m hmm-tumor -o sample.segment.hmm_tumor.cns
cnvkit call sample.segment.hmm_tumor.cns -t=-1.1,-0.5,0.4,0.9 -o sample.call.hmm_tumor.cns
cnvkit scatter sample.fix.cnr -s sample.segment.hmm_tumor.cns -g ERBB2 -o sample.cnv.scatter.ERBB2.png
The png as follow:
![image](https://github.com/etal/cnvkit/assets/82378273/f1e8b5c5-9abb-4baa-97fb-2d774c44e78e)

And the log2-ratio in cns seem to be lower than the log2-ratio in cnr , and i notices that "cnvkit.py segment " have the option -t  to set the threshold , so my question is:
1.The default value for the -t .
2.The ERBB2 log2-ratio in cns may be affected by the upstream and downstream gene log2-ratio, how can i set the threshold to make the segment in each gene。
And here is the cnr file and cns file :
[sample.call.hmm_tumor.cns.txt](https://github.com/etal/cnvkit/files/12889252/sample.call.hmm_tumor.cns.txt)
[sample.fix.cnr.txt](https://github.com/etal/cnvkit/files/12889253/sample.fix.cnr.txt)

Hope your reply.
Thanks!

**Comment by gevro** [comment]
Hello, I am also curious if anyone knows the default value for the segment command's threshold parameter?

**Comment by etal** [comment]
You can see the defaults in the help text:
```
cnvkit.py call -h
...
  -m {threshold,clonal,none}, --method {threshold,clonal,none}
                        Calling method. [Default: threshold]
  -t THRESHOLDS, --thresholds THRESHOLDS
                        Hard thresholds for calling each integer copy number, separated by commas. Use the '=' sign on the command line, e.g.: -t=-1,0,1 [Default: -1.1,-0.25,0.2,0.7]
  --ploidy PLOIDY       Ploidy of the sample cells. [Default: 2]
  --purity PURITY       Estimated tumor cell fraction, a.k.a. purity or cellularity.
```

There are some parameters in the `segment` command that affect how fine-grained the segmentation will be, but it's difficult to predict exactly what effect these will have; best to just try them and see.

Also consider using the `-m hmm` in the latest release and `-m haar` in the development version (source code and `devel` docker container). These will segment differently than the default CBS does.

As another fallback, you can use the `gainloss` command on the .cnr file alone to get the gene's average log2 without segmenting together with adjacent regions. For a big gene like EGFR, the gene average should be fairly reliable for spotting an amplification.
